File: connection-examples/lua/insert.lua
local utils = require("utils")
local driver = require "luasql.postgres"
local env = assert (driver.postgres())
local con = assert(env:connect("postgresql://MATERIALIZE_USERNAME:MATERIALIZE_PASSWORD@MATERIALIZE_HOST:6875/materialize?sslmode=require"))

assert (con:execute([[
CREATE TABLE IF NOT EXISTS countries (name TEXT, code TEXT);
]]))

local list = {
  { name="United States", code="US", },
  { name="Canada", code="CA", },
  { name="Mexico", code="MX", },
  { name="Germany", code="DE", },
}

for _, p in pairs (list) do
  local _ = assert (con:execute(string.format([[
    INSERT INTO countries
    VALUES ('%s', '%s')]], p.name, p.code)
  ))
end

con:commit()

for count in utils.rows(con, 'SELECT COUNT(*) FROM countries;') do
    print(string.format("%s", count))
end

File: connection-examples/php/insert.php
<?php
// Include the Postgres connection details
require 'connection.php';

$sql = 'INSERT INTO countries (name, code) VALUES (?, ?)';
$statement = $connection->prepare($sql);
$statement->execute(['United States', 'US']);
$statement->execute(['Canada', 'CA']);
$statement->execute(['Mexico', 'MX']);
$statement->execute(['Germany', 'DE']);

$countStmt = "SELECT COUNT(*) FROM countries";
$count = $connection->query($countStmt);
while (($row = $count->fetch(PDO::FETCH_ASSOC)) !== false) {
    var_dump($row);
}